### PR TITLE
Log out all session for a target

### PR DESCRIPTION
**Pull request by shuo-wu**
longhorn/longhorn#1920

### CODE REVIEW

**Comment by shuo-wu** [comment]
I am not sure how to make the 2 different volumes/ISCSI initiators use the same underlying Longhorn engine. But the root cause of this issue is that the target logout/cleanup function doesn't work if the IP address somehow gets changed.

**Comment by yasker** [comment]
@shuo-wu See the original issue. The user's engine instance manager was killed by OOMKiller, so it's essentially SIGKILL, without leaving a chance to the instance manager to clean up after itself. Then the engine instance manager IP changed during recreation, result in this situation.

File: iscsidev/iscsi.go [hunk 1]
```diff
@@ -209,12 +207,12 @@ func LogoutTarget(target string) error {
...
 		 */
 		for i := 0; i < RetryCounts; i++ {
-			if !iscsi.IsTargetDiscovered(ip, target, ne) {
+			if !iscsi.IsTargetDiscovered("", target, ne) {
 				err = nil
 				break
 			}
 
-			err = iscsi.DeleteDiscoveredTarget(ip, target, ne)
+			err = iscsi.DeleteDiscoveredTarget("", target, ne)
```
Comment (yasker) [inline iscsidev/iscsi.go:215]: Is this going to logout all the existing sessions for the target? If there is more than one, just in case.

Comment (shuo-wu) [reply iscsidev/iscsi.go:215]: Yes. The test mentioned in the original issue is passed, which means this updated function can clean up all sessions for the target.

**Comment by shuo-wu** [comment]
> @shuo-wu See the original issue. The user's engine instance manager was killed by OOMKiller, so it's essentially SIGKILL, without leaving a chance to the instance manager to clean up after itself. Then the engine instance manager IP changed during recreation, result in this situation.

I see. But how can we trigger IP change after the engine manager SIGKILL?

**Comment by yasker** [comment]
We can kill the instance and force delete the pod.

**Comment by shuo-wu** [comment]
The test is updated and passed.

**Review by yasker — APPROVED** [review]

**Pull request merged**